Modules/_xxtestfuzz/fuzzer.c

```diff
@@ -142,6 +142,74 @@ static int fuzz_struct_unpack(const char* data, size_t size) {
 }
 
 
+PyObject* pickle_loads_method = NULL;
+PyObject* pickle_error = NULL;
+PyObject* pickling_error = NULL;
+PyObject* unpickling_error = NULL;
+/* Called by LLVMFuzzerTestOneInput for initialization */
+static int init_pickle_loads() {
+    /* Import pickle.loads */
+    PyObject* pickle_module = PyImport_ImportModule("pickle");
+    if (pickle_module == NULL) {
+        return 0;
+    }
+    pickle_error = PyObject_GetAttrString(pickle_module, "PickleError");
+    if (pickle_error == NULL) {
+        return 0;
+    }
+    pickling_error = PyObject_GetAttrString(pickle_module, "PicklingError");
+    if (pickling_error == NULL) {
+        return 0;
+    }
+    unpickling_error = PyObject_GetAttrString(pickle_module, "UnpicklingError");
+    if (unpickling_error == NULL) {
+        return 0;
+    }
+    pickle_loads_method = PyObject_GetAttrString(pickle_module, "loads");
+    return pickle_loads_method != NULL;
+}
+
+#define MAX_PICKLE_TEST_SIZE 0x10000
+/* Fuzz pickle.loads(x) */
+static int fuzz_pickle_loads(const char* data, size_t size) {
+    if (size > MAX_PICKLE_TEST_SIZE) {
+        return 0;
+    }
+    PyObject* input_bytes = PyBytes_FromStringAndSize(data, size);
+    if (input_bytes == NULL) {
+        return 0;
+    }
+    PyObject* parsed = PyObject_CallOneArg(pickle_loads_method, input_bytes);
+    if (parsed == NULL &&
+            (PyErr_ExceptionMatches(PyExc_ValueError) ||
+            PyErr_ExceptionMatches(PyExc_AttributeError) ||
+            PyErr_ExceptionMatches(PyExc_KeyError) ||
+            PyErr_ExceptionMatches(PyExc_TypeError) ||
+            PyErr_ExceptionMatches(PyExc_OverflowError) ||
+            PyErr_ExceptionMatches(PyExc_EOFError) ||
+            PyErr_ExceptionMatches(PyExc_MemoryError) ||
+            PyErr_ExceptionMatches(PyExc_ModuleNotFoundError) ||
+            PyErr_ExceptionMatches(PyExc_IndexError) ||
+            PyErr_ExceptionMatches(PyExc_RecursionError) ||
+            PyErr_ExceptionMatches(PyExc_UnicodeDecodeError))) 
+    {
+        PyErr_Clear();
+    }
+
+    if (parsed == NULL && (
+           PyErr_ExceptionMatches(pickle_error) ||
+           PyErr_ExceptionMatches(pickling_error) ||
+           PyErr_ExceptionMatches(unpickling_error)
```

Comment:
Why not simply `PyErr_ExceptionMatches(PyExc_Exception)`? We don't really care which exact exception is raised.

Reply:
other fuzzers in this module also work like this, but if we're going to list exceptions to ignore, we should at least add comments as to why for each or at least each class.

I assume this being a pickle.loads test is really looking for process crashes.

unfortunately... I expect the results of this as a fuzzer to be pretty bad, pickle is explicitly documented as not being suitable for untrusted data.  it may execute arbitrary code or crash in that situation.  i asked this broader question on the bug.